optaplanner-core/src/main/java/org/optaplanner/core/api/score/AbstractScore.java

```diff
@@ -31,7 +31,7 @@
  * @see Score
  * @see HardSoftScore
  */
-public abstract class AbstractScore<S extends Score> implements Score<S>,
+public abstract class AbstractScore<S extends Score<S>> implements Score<S>,
```

Comment:
Technically, this is not necessary, but I did it anyway. It's the right thing to do.